tests/xtimer_mutex_lock_timeout/main.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2019 Freie Universitaet Berlin,
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       testing xtimer_mutex_lock_timeout function
+ *
+ *
+ * @author      Julian Holzwarth <julian.holzwarth@fu-berlin.de>
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "shell.h"
+#include "xtimer.h"
+
+/**
+ * Foward declarations
+ */
+static int cmd_test_xtimer_mutex_lock_timeout_greater_backoff(int argc,
+                                                              char **argv);
+static int cmd_test_xtimer_mutex_lock_timeout_greater_backoff_locked(int argc,
+                                                                     char **argv);
+
+/**
+ * @brief   List of command for this application.
+ */
+static const shell_command_t shell_commands[] = {
+    { "mutex_timeout_n_spin_unlocked", "unlocked mutex and without spinning",
+      cmd_test_xtimer_mutex_lock_timeout_greater_backoff, },
+    { "mutex_timeout_n_spin_locked", "locked mutex and without spinning",
+      cmd_test_xtimer_mutex_lock_timeout_greater_backoff_locked, },
+    { NULL, NULL, NULL }
+};
+
+/**
+ * @brief   shell command to test xtimer_mutex_lock_timeout
+ *
+ * the mutex is not locked before the function call and
+ * the timeout is greater than XTIMER_BACKOFF (no spinning)
```

Comment:
Please update in the direction, "the timeout callback will be removed before triggering" it would explain more the case.